src/lib/auth/AuthProviders/OAuthAuthenticator/index.ts

```diff
@@ -0,0 +1,122 @@
+import axios from 'axios'
+import { createHash } from 'crypto'
+import isWsl from 'is-wsl'
+import jwt from 'jsonwebtoken'
+import opn from 'opn'
+import { join } from 'path'
+import { logger } from '../../../../api'
+import { VTEXID } from '../../../../api/clients/IOClients/external/VTEXID'
+import { storeUrl } from '../../../../api/storeUrl'
+import { ColorifyConstants } from '../../../constants/Colors'
+import { randomCryptoString } from '../../../utils/randomCryptoString'
+import { spawnUnblockingChildProcess } from '../../../utils/spawnUnblockingChildProcess'
+import { AuthProviderBase } from '../AuthProviderBase'
+import { LoginServer } from './LoginServer'
+
+export class OAuthAuthenticator extends AuthProviderBase {
+  public static readonly AUTH_TYPE = 'oauth'
+  private static readonly SECRET_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~'
+
+  private static ADMIN_LOGIN_URL_PATH = '/_v/segment/admin-login/v1/login'
+  private static FALLBACK_AUTH_SERVER_LOGIN_URL_PATH = '/_v/private/auth-server/v1/login'
+
+  public async login(account: string) {
+    const secret = randomCryptoString(128, OAuthAuthenticator.SECRET_ALPHABET)
+    const secretHash = this.hashSecret(secret)
+    const vtexId = VTEXID.createClient({ account })
+
+    const loginServer = await LoginServer.create({
+      account,
+      secret,
+    })
+
+    try {
+      const loginState = await vtexId.startToolbeltLogin({
+        account,
+        secretHash,
+        loopbackUrl: loginServer.loginCallbackUrl,
+      })
+
+      loginServer.setLoginState(loginState)
+
+      const url = await this.loginUrl(account, loginState)
+      opn(url, { wait: false })
+
+      if (isWsl) {
+        logger.warn(
+          "We noticed you're using WSL, in which case you may face login issues depending on your WSL version."
+        )
+
+        logger.warn(
+          `If you do, make sure your Windows is up to date and try again (${ColorifyConstants.URL_INTERACTIVE(
+            'https://support.microsoft.com/en-us/help/4027667/windows-10-update'
+          )}).`
+        )
+
+        logger.warn(
+          `In case login errors persist after updating please create an issue on ${ColorifyConstants.URL_INTERACTIVE(
+            'https://github.com/vtex/toolbelt/issues'
+          )}. We'll promptly help you finding a solution.`
```

Comment:
This way sounds better:
```suggestion
          )}. We'll promptly help you find a solution.`
```
but I'm not 100% sure 🤔, what do you think?

Reply:
I think it sounds better indeed